### PR TITLE
fix: enable processor options to be passed processors

### PR DESCRIPTION
**Pull request by artur-ciocanu**
**Description**
This is a small refactoring PR to ensure that all the `processors` have access to `processorOptions` which is passed via `process(...)` method.

Also it adds `interpreterOptions` to `processorOptions` under `interpreter` and it ensures that `interpreter.interpret(...)` receives `interpreterOptions` that is referenced from `processorOptions`.

**Related issue(s)**
https://github.com/asyncapi/modelina/issues/916

### CODE REVIEW

**Comment by artur-ciocanu** [comment]
@jonaslagoni @ron-debajyoti @magicmatatjahu could someone provide some pointers on how to fix the "Blackbox testing". Thank you.

**Comment by jonaslagoni** [comment]
> @jonaslagoni @ron-debajyoti @magicmatatjahu could someone provide some pointers on how to fix the "Blackbox testing". Thank you.

You can ignore them, for now and just focus on the other CI checks 🙂 I am trying to solve them here #905

**Review by jonaslagoni — COMMENTED** [review]
@artur-ciocanu I am gonna review this later today or tomorrow.

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
For your purpose in regards to https://github.com/asyncapi/modelina/issues/916 do you mind clarifying why you want this change instead of letting the `process` functions carry the options around?

Cause with this change you end up creating a new instance per generator, instead of relying on the static instance. And I don't see the clear benefits, especially since you HAVE to provide it here anyway:  https://github.com/asyncapi/modelina/blob/21ce51034f8c0e16bd7eea21f29dc695bc1b0e42/src/processors/JsonSchemaInputProcessor.ts#L313

And so forth for all the processors that call the function. So I don't really see the need to change the architecture of this without further discussing it in an issue that proposes the change 🙂

**Comment by artur-ciocanu** [comment]
@jonaslagoni thanks a lot for your feedback and review. I have changed the approach to pass `processorOptions` to `processor.process()` and ensure that `interpreterOptions` is passed to `JsonSchemaInputProcessor.convertSchemaToCommonModel(...)`.

I have also changed the PR description, so it reflects the implementation.

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Just have one comment 🙂

File: src/models/ProcessorOptions.ts [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import { ParserOptions } from '@asyncapi/parser';
+import { InterpreterOptions } from 'interpreter/Interpreter';
```
[the remark this answers is not in the document]
Comment (artur-ciocanu) [reply src/models/ProcessorOptions.ts]: Good call. Fixed.

**Comment by artur-ciocanu** [comment]
@jonaslagoni I have addressed your comment. Please let me know if everything is looking good. Thank you!

**Review by jonaslagoni — APPROVED** [review]
LGTM 👍 

Changing the title to `fix` because this actually enables options to be passed along from the generators, so it's not so much a refactor 🙂

**Pull request merged**